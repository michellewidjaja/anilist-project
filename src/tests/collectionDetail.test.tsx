import React from 'react';
import '@testing-library/jest-dom';
import CollectionDetail from '../pages/collectionDetail';
import {render, screen, waitFor} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom';

test('renders collection detail to match snapshot', () => {
  const tree = render(
    <MemoryRouter initialEntries={['?collection=coll-1']}>
      <CollectionDetail />
    </MemoryRouter>);
  
  expect(tree).toMatchSnapshot();
});

test('show collection detail successfully', () => {
  const tree = render(
    <MemoryRouter initialEntries={['?collection=coll-1']}>
      <CollectionDetail />
    </MemoryRouter>);
  
  expect(screen.getByText('Collection coll-1')).toBeInTheDocument();
});