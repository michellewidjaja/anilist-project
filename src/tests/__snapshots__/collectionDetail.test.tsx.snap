// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders collection detail to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-13ya3ww-CollectionDetail"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-left "
          data-icon="arrow-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
        Back
      </div>
      <div
        class="css-8mxxdf-CollectionDetail"
      >
        Collection 
        coll1
      </div>
      <div
        class="css-1w2z1mm-CollectionDetail"
      >
        <div>
          There is no collection data
        </div>
        <div
          class="css-1marw79"
        >
          <div
            class="css-8cy5fh"
          >
            <div
              class="css-fky9qs-ModalConfirmation"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark "
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 384 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="css-89cmbv-ModalConfirmation"
            >
              Delete undefined from this collection?
            </div>
            <div
              class="css-qgy5z2-ModalConfirmation"
            >
              <div
                class="css-1moqzbc"
              >
                No
              </div>
              <div
                class="css-1tewy82"
              >
                Confirm
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-13ya3ww-CollectionDetail"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
        />
      </svg>
      Back
    </div>
    <div
      class="css-8mxxdf-CollectionDetail"
    >
      Collection 
      coll1
    </div>
    <div
      class="css-1w2z1mm-CollectionDetail"
    >
      <div>
        There is no collection data
      </div>
      <div
        class="css-1marw79"
      >
        <div
          class="css-8cy5fh"
        >
          <div
            class="css-fky9qs-ModalConfirmation"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="css-89cmbv-ModalConfirmation"
          >
            Delete undefined from this collection?
          </div>
          <div
            class="css-qgy5z2-ModalConfirmation"
          >
            <div
              class="css-1moqzbc"
            >
              No
            </div>
            <div
              class="css-1tewy82"
            >
              Confirm
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
